src/maker/api.rs

```diff
@@ -589,6 +589,12 @@ pub(crate) fn check_for_broadcasted_contracts(maker: Arc<Maker>) -> Result<(), M
                                 next_internal_address,
                                 DEFAULT_TX_FEE_RATE,
                             )?;
+                            let hash_lock_spend = maker.wallet.read()?.create_hashlock_spend(
+                                ic_sc,
+                                next_internal_address,
+                                DEFAULT_TX_FEE_RATE,
+                            )?;
+
```

Comment:
One of the things that can happen here is that this maker might not have received the Hash Preimage but the next maker in the swap has received the Hash Preimage and claims the outgoing swapcoins via Hashlock. In this scenario the maker will fail to create the hashlock spend. In order to prevent this, we need to keep watching the outgoing contract(when we don't have the Hash Preimage) even after it has been broadcasted and look for any transaction that spends from that broadcasted outgoing contract transaction, extract the hash preimage from that spending transaction, then claim the incoming contract using that hash preimage.

Reply:
This involves: 
1. writing logic for watching a spending transaction for an already broadcasted outgoing contract
2. writing a utility function to extract the hash preimage from the spending transaction
3. adding logic to use that hash preimage to create a spending transaction for the incoming contract